Modules/_sqlite/module.c

```diff
@@ -236,14 +236,17 @@ PyDoc_STRVAR(enable_callback_tracebacks_doc,
 \n\
 Enable or disable callback functions throwing errors to stderr.");
 
-static void converters_init(PyObject* dict)
+static void converters_init(PyObject* module)
 {
     _pysqlite_converters = PyDict_New();
     if (!_pysqlite_converters) {
         return;
     }
 
```

Comment:
```suggestion
Py_INCREF(_pysqlite_converters);
```

Reply:
`PyDict_New` returns a new reference, so this should not be necessary, no?

Reply:
Anyway, I've applied the suggested change, but I still don't understand why. From the docs:
```
PyObject *t;

t = PyTuple_New(3);
PyTuple_SetItem(t, 0, PyLong_FromLong(1L));
PyTuple_SetItem(t, 1, PyLong_FromLong(2L));
PyTuple_SetItem(t, 2, PyUnicode_FromString("three"));
```

The above is possible because, as far as I can understand:
1) `PyLong_FromLong` and `PyUnicode_FromString` returns _a new reference_
2) `PyTuple_SetItem` _steals_ the new reference

If `PyTuple_SetItem` _borrowed_ the reference, the above would not be possible.

Further: according to the docs, `PyModule_AddObject` _steals_ a reference when successful, so if we assume (hypothetically) that `PyModule_AddObject` never failed, we should be able to write:
```
PyModule_AddObject(m, "some_number", PyLong_FromLong(...));
PyModule_AddObject(m, "some_exception", PyErr_NewException(...));
PyModule_AddObject(m, "some_dict", PyDict_New(...));
```

However, since `PyModule_AddObject` only steals a reference when successful, we have to clean up if it fails, thus:
```
PyObject *d = PyDict_New(); // returns new reference
int ret = PyModule_AddObject(m, "my_dict", d);
if (ret) { ; } // previous reference was stolen, we're free!
else { Py_DECREF(d); } // reference was not stolen; we have to clean up!
```

Either I completely misunderstand [reference count details](https://docs.python.org/3.10/c-api/intro.html#reference-count-details), or the `*_New()` functions stopped returning new references, or `PyModule_AddObject` does not steal references anymore.

I'd appreciate if you could enlighten me, @corona10 :)

Reply:
@erlend-aasland 
Oh, you are right, my review was wrong.
Can you revert the

Py_INCREF(_pysqlite_converters);
Py_INCREF(psyco_adapters);
Py_INCREF(exc);

That I pointed out? The latest documents are missing `Return value: New reference.` which should noted.
I will create this issue after I check the status.

Reply:
@corona10 Ah, thanks :) I was re-reading and re-reading the docs, and I just couldn't get it to make sense! I think some of the examples in https://docs.python.org/3.10/extending/extending.html should be fixed as well :) I'll revert the problematic commits and request your review again.

Reply:
Reverted with 5ac7298 and 94e66e1!

`./python.exe -m test -F -r -j1  -R 3:10 test_sqlite` reports success; no leaks

Reply:
PTAL :)